lib/ber/reader.js

```diff
@@ -225,25 +226,47 @@ Reader.prototype._readTag = function(tag) {
   if (o === null)
     return null;
 
-  if (this.length > 4)
+  if (this.length > 8)
     throw newInvalidAsn1Error('Integer too long: ' + this.length);
 
   if (this.length > this._size - o)
     return null;
   this._offset = o;
 
-  var fb = this._buf[this._offset];
-  var value = 0;
-
-  for (var i = 0; i < this.length; i++) {
-    value <<= 8;
-    value |= (this._buf[this._offset++] & 0xff);
-  }
+  if (!this.length)
+    return -1;
```

Comment:
Your semantics here are "if length is 0, return -1". I don't see corresponding semantics in the original. What happened if length was 0 in the original? I would veer towards retaining that behaviour (you want to change _one_ thing: add support for long ints; don't change other things while you're at it)